integrated_channels/integrated_channel/exporters/content_metadata.py

```diff
@@ -596,25 +611,21 @@ def _transform_item(self, content_metadata_item):
             # Look for transformer functions defined on subclasses.
             # Favor content type-specific functions.
             transformer = (
-                getattr(
-                    self,
-                    'transform_{content_type}_{edx_data_schema_key}'.format(
-                        content_type=content_metadata_type,
-                        edx_data_schema_key=edx_data_schema_key
-                    ),
-                    None
-                )
+                getattr(self, f'transform_{content_metadata_type}_{edx_data_schema_key}', None)
                 or
-                getattr(
-                    self,
-                    'transform_{edx_data_schema_key}'.format(
-                        edx_data_schema_key=edx_data_schema_key
-                    ),
-                    None
-                )
+                getattr(self, f'transform_{edx_data_schema_key}', None)
             )
+
+            transformer_for_action = (
+                getattr(self, f'transform_for_action_{content_metadata_type}_{edx_data_schema_key}', None)
```

Comment:
quick question: In what instance will this be used? Where is this attribute defined?

Reply:
The intention's to go into the channel specific content metadata exporter and use the defined mappings to tailor and sanitize the code (see https://github.com/openedx/edx-enterprise/blob/master/integrated_channels/sap_success_factors/exporters/content_metadata.py). There we have two different types of mapping methods, `transform_{attribute}` and `transform_{content_type}_{attribute}` because some external platforms care about and want to distinguish between metadata depending on what type of content it is, ie SAP wants different titles for course runs than for course and programs. In this case we're supporting the old implementation while also allowing for the caller to specify more detail/actions.  `transform_{content_type}_{attribute}` isn't currently being used by any channel yet but I wanted to add the functionality as we know channels do this content specific mapping already and we may want to support it in an upgrade in the future.